contracts/protocol/facets/Offer.sol

```diff
@@ -419,7 +422,10 @@ contract OfferFacet is Context, OfferErrors, Access, FundsManager, IOfferEvents
             EntityLib.validateSellerAssistantOrFacilitator(sellerId, offer.facilitatorId);
             handleBosonSellerDeposit(sellerId, exchangeToken, offer.sellerDeposit);
 
-            // WrapType wrapType = _wrapType;
+            if (_wrapType != FermionTypes.WrapType.SELF_SALE && offer.sellerDeposit == 0 && msg.value != 0) {
```

Comment:
Not clear to me why we exclude the case where offer.sellerDeposit != 0.
If an offer.exchangeToken is an ERC20, it doesn't make sense to send native funds on the transaction, so I think it should revert whatever the value of the sellerDeposit.

Reply:
When `offer.sellerDeposit != 0`, the checks are done in `handleBosonSellerDeposit`.

At least, they are when `availableFunds < _sellerDeposit`, but they were missing for the case `availableFunds >= _sellerDeposit`. I added it now, thanks for asking.